### PR TITLE
Hasura GraphQL

### PR DESCRIPTION
**Pull request by rushsteve1**
This PR adds the Hasura GraphQL endpoint and it's relevant configuration, as well as alters migrations to match Hasura's CLI migration tool.

**Note that the migrations in this PR reflect the REAL STATE OF THE DATABASE RIGHT NOW**

Outstanding issues:
- Documentation/README updates
- Example updates

### CODE REVIEW

**Comment by vcfxb** [comment]
This PR should probably remove or update the example too.

File: migrations/1610912853000_create_roles/down.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+-- The order of these must be the reverse of above
```
Comment (vcfxb) [inline migrations/1610912853000_create_roles/down.sql]: It's in the other file now, not "above" anymore.

**Review by vcfxb — COMMENTED** [review]
Looks mostly good -- some wording on some files might need to be updated to reflect the separation of UP and DOWN migrations. The python example file should also be updated to work or be removed.

**Review by vcfxb — APPROVED** [review]
Looks good to me :)

**Review by Apexal — APPROVED** [review]

**Pull request merged**